bug: hooks fail in monorepo subdirectories due to relative paths

## Description

When users install ClaudeKit in a monorepo root and then start a Claude session from a subdirectory, hooks fail with `MODULE_NOT_FOUND` because paths in `settings.json` are relative to CWD, not to where `.claude/` exists.

## Steps to Reproduce

1. Initialize ClaudeKit in monorepo root:
   ```bash
   cd ~/projects/my-monorepo
   ck init
   ```

2. Start Claude session from subdirectory:
   ```bash
   cd ~/projects/my-monorepo/frontend
   claude
   ```

3. **Result**: Hook errors:
   ```
   Error: Cannot find module '/Users/user/projects/my-monorepo/frontend/.claude/hooks/dev-rules-reminder.cjs'
   ```

## Root Cause

- `settings.json` contains relative paths: `node .claude/hooks/dev-rules-reminder.cjs`
- Claude Code resolves paths from CWD, not from where `.claude/` directory exists
- Global install works because CLI transforms to `$HOME/.claude/...` (Unix) or `%USERPROFILE%/.claude/...` (Windows)
- Local install does NOT transform paths

## Expected Behavior

Local install should transform relative `.claude/` paths to use `$CLAUDE_PROJECT_DIR`:
```json
"command": "node \"$CLAUDE_PROJECT_DIR\"/.claude/hooks/dev-rules-reminder.cjs"
```

Claude Code sets `$CLAUDE_PROJECT_DIR` to the directory containing `.claude/` when spawning hooks.

## Proposed Fix

Modify `processSettingsJson()` in `src/lib/merge.ts` to transform paths for both local and global installs:

| Mode | Input | Output |
|------|-------|--------|
| Global (Unix) | `node .claude/...` | `node $HOME/.claude/...` |
| Global (Windows) | `node .claude/...` | `node %USERPROFILE%/.claude/...` |
| Local (Unix) | `node .claude/...` | `node "$CLAUDE_PROJECT_DIR"/.claude/...` |
| Local (Windows) | `node .claude/...` | `node "%CLAUDE_PROJECT_DIR%"/.claude/...` |

This keeps the template as canonical source with relative paths, and CLI handles all environment-specific transformations.

## Environment

- ClaudeKit CLI: v3.5.0
- Node.js: v22.18.0+
- Affected: Local installs in monorepo structures

## Related

- Global path transformer already exists: `src/lib/global-path-transformer.ts`
- `processSettingsJson()` already handles global mode transformation

--- a/src/lib/merge.ts
+++ b/src/lib/merge.ts
@@ -129,7 +129,11 @@ export class FileMerger {
 			}
 
 			// Special handling for settings.json - convert env var syntax for cross-platform
-			if (normalizedRelativePath === "settings.json") {
+			// Handle both root settings.json and .claude/settings.json (depends on source structure)
+			if (
+				normalizedRelativePath === "settings.json" ||
+				normalizedRelativePath === ".claude/settings.json"
+			) {
 				await this.processSettingsJson(file, destPath);
 				this.trackInstalledFile(normalizedRelativePath);
 				copiedCount++;
@@ -145,12 +149,14 @@ export class FileMerger {
 	}
 
 	/**
-	 * Process settings.json file and convert Unix-style env vars to Windows syntax
+	 * Process settings.json file and transform paths for cross-platform compatibility
 	 *
-	 * Cross-platform compatibility:
-	 * - Global mode: $CLAUDE_PROJECT_DIR → $HOME (Unix) or %USERPROFILE% (Windows)
-	 * - Local mode on Windows: $VAR → %VAR% (convert syntax only)
-	 * - Local mode on Unix: No changes needed
+	 * Path transformation rules:
+	 * - Global mode: .claude/ → $HOME/.claude/ (Unix) or %USERPROFILE%/.claude/ (Windows)
+	 * - Local mode: .claude/ → "$CLAUDE_PROJECT_DIR"/.claude/ (Unix) or "%CLAUDE_PROJECT_DIR%"/.claude/ (Windows)
+	 *
+	 * This enables monorepo support where users can start Claude sessions from subdirectories
+	 * while hooks/statusline scripts resolve correctly relative to where .claude/ exists.
 	 */
 	private async processSettingsJson(sourceFile: string, destFile: string): Promise<void> {
 		try {
@@ -160,28 +166,27 @@ export class FileMerger {
 			let processedContent = content;
 
 			if (this.isGlobal) {
-				// Global mode: Replace $CLAUDE_PROJECT_DIR with home directory
+				// Global mode: Replace relative .claude/ paths with home directory
 				const homeVar = isWindows ? "%USERPROFILE%" : "$HOME";
-				processedContent = content.replace(/\$CLAUDE_PROJECT_DIR/g, homeVar);
+				processedContent = this.transformClaudePaths(content, homeVar);
 
 				if (processedContent !== content) {
 					logger.debug(
-						`Replaced $CLAUDE_PROJECT_DIR with ${homeVar} in settings.json for global installation`,
+						`Transformed .claude/ paths to ${homeVar}/.claude/ in settings.json for global installation`,
 					);
 				}
-			} else if (isWindows) {
-				// Local mode on Windows: Convert $VAR syntax to %VAR% syntax
-				// $CLAUDE_PROJECT_DIR → %CLAUDE_PROJECT_DIR%
-				// $HOME → %USERPROFILE%
-				processedContent = content
-					.replace(/\$CLAUDE_PROJECT_DIR/g, "%CLAUDE_PROJECT_DIR%")
-					.replace(/\$HOME/g, "%USERPROFILE%");
+			} else {
+				// Local mode: Replace relative .claude/ paths with $CLAUDE_PROJECT_DIR
+				// This enables monorepo support (running from subdirectories)
+				const projectDirVar = isWindows ? '"%CLAUDE_PROJECT_DIR%"' : '"$CLAUDE_PROJECT_DIR"';
+				processedContent = this.transformClaudePaths(content, projectDirVar);
 
 				if (processedContent !== content) {
-					logger.debug("Converted Unix env var syntax to Windows syntax in settings.json");
+					logger.debug(
+						`Transformed .claude/ paths to ${projectDirVar}/.claude/ in settings.json for local installation`,
+					);
 				}
 			}
-			// Local mode on Unix: No changes needed, $CLAUDE_PROJECT_DIR works as-is
 
 			// Write the processed content to destination
 			await writeFile(destFile, processedContent, "utf-8");
@@ -190,6 +195,44 @@ export class FileMerger {
 			// Fallback to direct copy if processing fails
 			await copy(sourceFile, destFile, { overwrite: true });
 		}
+	}
+
+	/**
+	 * Transform relative .claude/ paths to use a prefix variable
+	 *
+	 * Handles patterns like:
+	 * - "node .claude/hooks/..." → "node \"$PREFIX\"/.claude/hooks/..."
+	 * - "node ./.claude/hooks/..." → "node \"$PREFIX\"/.claude/hooks/..."
+	 *
+	 * The quotes around the env var are escaped for JSON and ensure paths with
+	 * spaces work correctly when the shell expands the variable.
+	 *
+	 * Only transforms paths in command contexts (after "node ", in quotes, etc.)
+	 * to avoid transforming documentation or other non-path content.
+	 */
+	private transformClaudePaths(content: string, prefix: string): string {
+		let transformed = content;
+
+		// Escape quotes for JSON if prefix contains quotes (local mode)
+		// e.g., "$CLAUDE_PROJECT_DIR" → \"$CLAUDE_PROJECT_DIR\"
+		const jsonSafePrefix = prefix.includes('"') ? prefix.replace(/"/g, '\\"') : prefix;
+
+		// Pattern 1: "node .claude/" or "node ./.claude/" - common hook command pattern
+		// Matches: "node .claude/..." or "node ./.claude/..."
+		transformed = transformed.replace(
+			/(node\s+)(?:\.\/)?\.claude\//g,
+			`$1${jsonSafePrefix}/.claude/`,
+		);
+
+		// Pattern 2: Already has $CLAUDE_PROJECT_DIR - replace with appropriate prefix
+		// This handles templates that already use the variable
+		if (prefix.includes("HOME") || prefix.includes("USERPROFILE")) {
+			// Global mode: $CLAUDE_PROJECT_DIR → $HOME or %USERPROFILE%
+			transformed = transformed.replace(/\$CLAUDE_PROJECT_DIR/g, prefix);
+			transformed = transformed.replace(/%CLAUDE_PROJECT_DIR%/g, prefix);
+		}
+
+		return transformed;
 	}
 
 	/**

--- a/tests/lib/merge.test.ts
+++ b/tests/lib/merge.test.ts
@@ -1316,7 +1316,298 @@ describe("FileMerger", () => {
 			// Empty directories shouldn't create entries
 			expect(installedFiles).toEqual([]);
 		});
+	});
 
+	describe("local mode: monorepo path transformation", () => {
+		test("should transform relative .claude/ paths to $CLAUDE_PROJECT_DIR on Unix (local mode)", async () => {
+			// Create settings.json with relative hook paths (as in claudekit-engineer template)
+			const settingsContent = JSON.stringify(
+				{
+					statusLine: {
+						type: "command",
+						command: "node .claude/statusline.cjs",
+					},
+					hooks: {
+						UserPromptSubmit: [
+							{
+								hooks: [
+									{
+										type: "command",
+										command: "node .claude/hooks/dev-rules-reminder.cjs",
+									},
+								],
+							},
+						],
+					},
+				},
+				null,
+				2,
+			);
+
+			await writeFile(join(testSourceDir, "settings.json"), settingsContent);
+
+			// Mock Unix platform
+			const originalPlatform = process.platform;
+			Object.defineProperty(process, "platform", {
+				value: "linux",
+				configurable: true,
+			});
+
+			try {
+				// Local mode (isGlobal = false by default)
+				await merger.merge(testSourceDir, testDestDir, true);
+
+				const destContent = await Bun.file(join(testDestDir, "settings.json")).text();
+				const destJson = JSON.parse(destContent);
+
+				// Verify paths transformed to use $CLAUDE_PROJECT_DIR (quotes are unescaped after JSON parse)
+				expect(destJson.statusLine.command).toBe(
+					'node "$CLAUDE_PROJECT_DIR"/.claude/statusline.cjs',
+				);
+				expect(destJson.hooks.UserPromptSubmit[0].hooks[0].command).toBe(
+					'node "$CLAUDE_PROJECT_DIR"/.claude/hooks/dev-rules-reminder.cjs',
+				);
+			} finally {
+				Object.defineProperty(process, "platform", {
+					value: originalPlatform,
+					configurable: true,
+				});
+			}
+		});
+
+		test("should transform relative .claude/ paths to %CLAUDE_PROJECT_DIR% on Windows (local mode)", async () => {
+			// Create settings.json with relative hook paths
+			const settingsContent = JSON.stringify(
+				{
+					statusLine: {
+						type: "command",
+						command: "node .claude/statusline.cjs",
+					},
+					hooks: {
+						PreToolUse: [
+							{
+								matcher: "Bash",
+								hooks: [
+									{
+										type: "command",
+										command: "node .claude/hooks/scout-block.cjs",
+									},
+								],
+							},
+						],
+					},
+				},
+				null,
+				2,
+			);
+
+			await writeFile(join(testSourceDir, "settings.json"), settingsContent);
+
+			// Mock Windows platform
+			const originalPlatform = process.platform;
+			Object.defineProperty(process, "platform", {
+				value: "win32",
+				configurable: true,
+			});
+
+			try {
+				// Local mode (isGlobal = false by default)
+				await merger.merge(testSourceDir, testDestDir, true);
+
+				const destContent = await Bun.file(join(testDestDir, "settings.json")).text();
+				const destJson = JSON.parse(destContent);
+
+				// Verify paths transformed to use %CLAUDE_PROJECT_DIR% (Windows syntax, quotes unescaped after parse)
+				expect(destJson.statusLine.command).toBe(
+					'node "%CLAUDE_PROJECT_DIR%"/.claude/statusline.cjs',
+				);
+				expect(destJson.hooks.PreToolUse[0].hooks[0].command).toBe(
+					'node "%CLAUDE_PROJECT_DIR%"/.claude/hooks/scout-block.cjs',
+				);
+			} finally {
+				Object.defineProperty(process, "platform", {
+					value: originalPlatform,
+					configurable: true,
+				});
+			}
+		});
+
+		test("should transform ./.claude/ paths (with leading dot-slash) on Unix", async () => {
+			// Create settings.json with ./.claude/ paths
+			const settingsContent = JSON.stringify(
+				{
+					statusLine: {
+						type: "command",
+						command: "node ./.claude/statusline.cjs",
+					},
+				},
+				null,
+				2,
+			);
+
+			await writeFile(join(testSourceDir, "settings.json"), settingsContent);
+
+			// Mock Unix platform
+			const originalPlatform = process.platform;
+			Object.defineProperty(process, "platform", {
+				value: "darwin",
+				configurable: true,
+			});
+
+			try {
+				await merger.merge(testSourceDir, testDestDir, true);
+
+				const destContent = await Bun.file(join(testDestDir, "settings.json")).text();
+				const destJson = JSON.parse(destContent);
+
+				// Should handle ./.claude/ same as .claude/ (quotes unescaped after JSON parse)
+				expect(destJson.statusLine.command).toBe(
+					'node "$CLAUDE_PROJECT_DIR"/.claude/statusline.cjs',
+				);
+			} finally {
+				Object.defineProperty(process, "platform", {
+					value: originalPlatform,
+					configurable: true,
+				});
+			}
+		});
+
+		test("should transform to $HOME in global mode (Unix)", async () => {
+			// Create settings.json with relative hook paths
+			const settingsContent = JSON.stringify(
+				{
+					hooks: {
+						UserPromptSubmit: [
+							{
+								hooks: [
+									{
+										type: "command",
+										command: "node .claude/hooks/dev-rules-reminder.cjs",
+									},
+								],
+							},
+						],
+					},
+				},
+				null,
+				2,
+			);
+
+			await writeFile(join(testSourceDir, "settings.json"), settingsContent);
+
+			// Mock Unix platform
+			const originalPlatform = process.platform;
+			Object.defineProperty(process, "platform", {
+				value: "linux",
+				configurable: true,
+			});
+
+			try {
+				merger.setGlobalFlag(true);
+				await merger.merge(testSourceDir, testDestDir, true);
+
+				const destContent = await Bun.file(join(testDestDir, "settings.json")).text();
+				const destJson = JSON.parse(destContent);
+
+				// Global mode should use $HOME
+				expect(destJson.hooks.UserPromptSubmit[0].hooks[0].command).toBe(
+					"node $HOME/.claude/hooks/dev-rules-reminder.cjs",
+				);
+			} finally {
+				Object.defineProperty(process, "platform", {
+					value: originalPlatform,
+					configurable: true,
+				});
+			}
+		});
+
+		test("should transform to %USERPROFILE% in global mode (Windows)", async () => {
+			// Create settings.json with relative hook paths
+			const settingsContent = JSON.stringify(
+				{
+					statusLine: {
+						type: "command",
+						command: "node .claude/statusline.cjs",
+					},
+				},
+				null,
+				2,
+			);
+
+			await writeFile(join(testSourceDir, "settings.json"), settingsContent);
+
+			// Mock Windows platform
+			const originalPlatform = process.platform;
+			Object.defineProperty(process, "platform", {
+				value: "win32",
+				configurable: true,
+			});
+
+			try {
+				merger.setGlobalFlag(true);
+				await merger.merge(testSourceDir, testDestDir, true);
+
+				const destContent = await Bun.file(join(testDestDir, "settings.json")).text();
+				const destJson = JSON.parse(destContent);
+
+				// Global mode should use %USERPROFILE%
+				expect(destJson.statusLine.command).toBe("node %USERPROFILE%/.claude/statusline.cjs");
+			} finally {
+				Object.defineProperty(process, "platform", {
+					value: originalPlatform,
+					configurable: true,
+				});
+			}
+		});
+
+		test("should not transform non-node commands", async () => {
+			// Create settings.json with non-node commands
+			const settingsContent = JSON.stringify(
+				{
+					hooks: {
+						UserPromptSubmit: [
+							{
+								hooks: [
+									{
+										type: "command",
+										command: "echo .claude/test",
+									},
+								],
+							},
+						],
+					},
+				},
+				null,
+				2,
+			);
+
+			await writeFile(join(testSourceDir, "settings.json"), settingsContent);
+
+			// Mock Unix platform
+			const originalPlatform = process.platform;
+			Object.defineProperty(process, "platform", {
+				value: "linux",
+				configurable: true,
+			});
+
+			try {
+				await merger.merge(testSourceDir, testDestDir, true);
+
+				const destContent = await Bun.file(join(testDestDir, "settings.json")).text();
+				const destJson = JSON.parse(destContent);
+
+				// Non-node commands should not be transformed
+				expect(destJson.hooks.UserPromptSubmit[0].hooks[0].command).toBe("echo .claude/test");
+			} finally {
+				Object.defineProperty(process, "platform", {
+					value: originalPlatform,
+					configurable: true,
+				});
+			}
+		});
+	});
+
+	describe("file tracking continued", () => {
 		test("should track multiple merges correctly (fresh merger each time)", async () => {
 			// First merge
 			await writeFile(join(testSourceDir, "file1.txt"), "content1");

